### ISSUE TITLE
Posting an Array of Objects as a body parameter not supported in OpenAPI v2

### ISSUE THREAD

**Issue by lafokamotion** [jhthorsen/mojolicious-plugin-openapi#240]
Similar issue to https://github.com/jhthorsen/mojolicious-plugin-openapi/issues/219

Except the body parameter is an array of objects containing a string as shown:
```
parameters:
  - name: echo
    in: body
    description: Text that should be echoed
    schema:
      type: array
      minItems: 1
      items:
        type: object
        required:
          - record
        properties:
          record:
            type: string
```

The idea being that we have a service that can take one or more items in a POST, shouldn't the spec be able to accept a json object such as:
```
[ { record => 'foo' }, { record => 'bar' } ]
```

However, when calling `$c->openapi->valid_input()`, the following message is received:
```
{
      errors => [
                  {
                    message => "Expected object - got string.",
                    path => "/echo/0",
                  },
                ],
      status => 400,
    }
```

This also is not an issue for OpenAPI v3 as I've tested the behavior there too.

Perl module versions:

- **Mojolicious**: 9.30
- **Mojolicious::Plugin::OpenAPI**: 5.07
- **JSON::Validator**: 5.12

**Comment by lafokamotion** [comment]
I was able to get this working again with an older version of JSON::Validator and Mojolicious::Plugin::OpenAPI:

Perl module versions:

    Mojolicious: 9.30
    Mojolicious::Plugin::OpenAPI: 3.41
    JSON::Validator: 4.25

JSON::Validator > 4.25 will return an error that object method "_resolve" can't be located via JSON::Validator::OpenAPI::Mojolicious.

Mojolicious::Plugin::OpenAPI < 4.00 will pass the code sample.

**Comment by jhthorsen** [comment]
Sorry for the late reply.

You're not posting JSON in your test case:

```diff
- $t->post_ok('/v1/echo' => $json)->status_is(200)->content_is($json);
+ $t->post_ok('/v1/echo', json => $json)->status_is(200)->content_is($json);
```

There's something strange going on with your setup, since JSON::Validator::OpenAPI::Mojolicious have been removed a long time ago: https://github.com/jhthorsen/mojolicious-plugin-openapi/commit/1adbaca4fcfe9684aba3bffa816cb21ad3680bff

**Comment by jhthorsen** [comment]
Regardless of my feedback, I found a bug which should be fixed in v5.08.

https://github.com/jhthorsen/json-validator/commit/96672ed1c3a4c05ee13783c3de15b7fc59c97be3

**Comment by lafokamotion** [comment]
Oh whoops, sorry about that. I'd probably been looking at my own code so long, i missed that syntax.
Thanks for your help though!
